tabbycat/participants/models.py

```diff
@@ -125,6 +128,11 @@ class Meta:
     def __str__(self):
         return str(self.name)
 
+    def get_public_name(self, tournament):
+        if tournament.pref('participant_code_names') == 'off':
+            return self.name
+        return self.code_name
```

Comment:
Passing in `tournament` is just a performance optimization, right? Since it can always just be inferred from `adj.tournament`? If so, are there any calls where this performance optimization doesn't do very much and we could do something like
```suggestion
    def get_public_name(self, tournament=None):
        if tournament is None:
          tournament = self.tournament
        if tournament.pref('participant_code_names') == 'off':
            return self.name
        return self.code_name
```
to simplify some calls?

(Or is it not a valid assumption that `adj.tournament is not None`? I can't remember if we still permit adj sharing - I have some vague recollection of deprecating this feature, and can't find `share_adjs` in the code base, but also `Adjudicator.tournament` is still nullable.)

(Also... I can't remember why I didn't use a function like this for team codes 😂)

Reply:
It can't really always be inferred by `adj.tournament` as it also affects speakers, who would then be `spk.team.tournament`.

Reply:
Ah, but then same idea applies? Either add logic to tell between the two (using duck typing or whatever) or add
``` python
    @property
    def tournament
        return self.team.tournament
```
to `Speaker`.

Anyway, I'm not too fussed about this, I can see how with two cases it becomes a little clunky. Can always add later if it becomes more bothersome.